cookbook/case_studies/ml_training/nlp_processing/word2vec_and_lda.py

```diff
@@ -0,0 +1,346 @@
+    """
+    Hyperparameters that can be used while training the LDA model.
+    """
+
+    num_topics: int = 5
+    alpha: str = "auto"
+    passes: int = 10
+    chunksize: int = 100
+    update_every: int = 1
+    random_state: int = 100
+
+
+# %%
+# Training
+# ========
+#
+# We initialize and train a Word2Vec model on the preprocessed corpus using hyperparameters defined.
+@task
+def train_word2vec_model(
+    training_data: List[List[str]], hyperparams: Word2VecModelHyperparams
+) -> model_file:
+
+    model = Word2Vec(
+        training_data,
+        min_count=hyperparams.min_count,
+        workers=hyperparams.workers,
+        vector_size=hyperparams.vector_size,
+        compute_loss=hyperparams.compute_loss,
+    )
+    training_loss = model.get_latest_training_loss()
+    logger.info(f"training loss: {training_loss}")
+    out_path = os.path.join(
+        flytekit.current_context().working_directory, "word2vec.model"
+    )
+    model.save(out_path)
+    return (out_path,)
+
+
+# %%
+# Next, we transform the documents to a vectorized form and compute the frequency of each word to generate a bag of
+# words corpus for the LDA model to train on. We also create a mapping from word IDs to words as input to
+# the LDA model for training.
+@task
+def train_lda_model(
+    corpus: List[List[str]], hyperparams: LDAModelHyperparams
+) -> model_file:
+    id2word = Dictionary(corpus)
+    bow_corpus = [id2word.doc2bow(doc) for doc in corpus]
+    id_words = [[(id2word[id], count) for id, count in line] for line in bow_corpus]
+    logger.info(f"sample of bag of words generated: {id_words[:2]}")
+    lda = LdaModel(
+        corpus=bow_corpus,
+        id2word=id2word,
+        num_topics=hyperparams.num_topics,
+        alpha=hyperparams.alpha,
+        passes=hyperparams.passes,
+        chunksize=hyperparams.chunksize,
+        update_every=hyperparams.update_every,
+        random_state=hyperparams.random_state,
+    )
+    logger.info(f"Shows words in each topic: {lda.print_topics(num_words=5)}")
+    out_path = os.path.join(flytekit.current_context().working_directory, "lda.model")
+    lda.save(out_path)
+    return (out_path,)
+
+
+# %%
+# Word Similarities
+# =================
+#
+# We deserialize the model from disk and compute the top 10 similar
+# words to the given word in the corpus (we will use the word `computer` when running
+# the workflow to output similar words). Note that since the model is trained
+# on a small corpus, some of the relations might not be clear.
+@task(cache_version="1.0", cache=True, limits=Resources(mem="600Mi"))
+def word_similarities(
+    model_ser: FlyteFile[MODELSER_NLP], word: str
+) -> Dict[str, float]:
+    model = Word2Vec.load(model_ser.download())
+    wv = model.wv
+    similar_words = wv.most_similar(word, topn=10)
+    logger.info(f"Word vector for {word}:{wv[word]}")
+    logger.info(f"Most similar words in corpus to {word}: {similar_words}")
+    return dict(similar_words)
```

Comment:
May I know why you're returning a dict if you aren't using it?

Reply:
so it shows up on flytedeck outputs  (like in screenshot) in case user wants to check directly from there.